eng/scripts/Cadl-Project-Generate.ps1

```diff
@@ -42,20 +42,27 @@ $npmWorkingDir = Resolve-Path $tempFolder/$innerFolder
 $mainCadlFile = If (Test-Path "$npmWorkingDir/client.cadl") { Resolve-Path "$npmWorkingDir/client.cadl" } Else { Resolve-Path "$npmWorkingDir/main.cadl"}
 $resolvedProjectDirectory = Resolve-Path "$ProjectDirectory/src"
 
+$HasEmitterOutputDir = $false;
 if (Test-Path "$npmWorkingDir/cadl-project.yaml") {
     $cadlProjectYaml = Get-Content -Path "$npmWorkingDir/cadl-project.yaml" -Raw
     $yml = ConvertFrom-YAML $cadlProjectYaml
-    if ( $yml["emitters"]["@azure-tools/cadl-csharp"]["sdk-folder"]) {
-        $yml["emitters"]["@azure-tools/cadl-csharp"]["sdk-folder"] = ""
+    if ($yml -And $yml["options"] -And $yml["options"]["@azure-tools/cadl-csharp"] -And $yml["options"]["@azure-tools/cadl-csharp"]["emitter-output-dir"]) {
+        $HasEmitterOutputDir = $true
     }
-    (ConvertTo-Yaml $yml) | Out-File "$npmWorkingDir/cadl-project.yaml"
 }
 
 try {
     Push-Location $npmWorkingDir
     NpmInstallForProject $npmWorkingDir
-    Write-Host("npx cadl compile $mainCadlFile --emit `"`@azure-tools/cadl-csharp`" --output-path `"$resolvedProjectDirectory`"")
-    npx cadl compile $mainCadlFile --emit "@azure-tools/cadl-csharp" --output-path "$resolvedProjectDirectory"
+    if ($HasEmitterOutputDir) {
+        $csharpSdkFolder = (Join-Path $ProjectDirectory ".." ".." "..")
+        $csharpSdkFolder = Resolve-Path $csharpSdkFolder
+        Write-Host("npx cadl compile $mainCadlFile --emit `"`@azure-tools/cadl-csharp`" --arg `"csharp-sdk-folder=$csharpSdkFolder`"")
+        npx cadl compile $mainCadlFile --emit "@azure-tools/cadl-csharp" --arg "csharp-sdk-folder=$csharpSdkFolder"
```

Comment:
Perhaps I've missed it but where are we overriding the version of cadl-csharp in the package.json file?

Reply:
hello @weshaggard we define the cadl-csharp version in the `eng\csharp-emitter-package.json`. When generate the SDK we will use this file as package.json to install the cadl-csharp emitter which have all the cadl dependency such as cadl compiler.

Reply:
Can you point me at in the script where we overwrite the package.json file?

Reply:
hello @weshaggard we override https://github.com/Azure/azure-sdk-for-net/blob/28936d2942627f84b123d43a47a4d1e49a07ab9d/eng/scripts/Cadl-Project-Generate.ps1#L23

Reply:
Nevermind I found it on line 23 of this script. While blindly replacing might work for the basic cases but we might need to move to a different model in the future because there might be other properties or settings in the package.json that we don't want to lose. We might need to do something similar to https://github.com/openapi-env-test/azure-sdk-for-java/blob/main/eng/mgmt/automation/generate_data.py#L67

cc @dw511214992

Reply:
yes. If it is needed, we will parse out the `@azure-tools/cadl-csharp` version out and install it directly in future.